extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheMethodValidator.java

```diff
@@ -0,0 +1,115 @@
+package io.quarkus.cache.deployment;
+
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE_ALL;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_LOAD;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_NAME_PARAMETER_NAME;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_STORE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.GENERATED_CACHE_NAME_PREFIX;
+
+import java.util.List;
+
+import javax.enterprise.inject.spi.DeploymentException;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+
+import io.quarkus.arc.processor.AnnotationStore;
+import io.quarkus.arc.processor.BeanInfo;
+
+/**
+ * @author Gwenneg Lepage
+ */
+public class CacheMethodValidator {
+
+    public static void validateAnnotations(AnnotationStore annotationStore, BeanInfo bean, MethodInfo method,
+            List<DeploymentException> exceptions) {
+
+        // TODO: @CacheKey should be validated here!
```

Comment:
I didn't find a way to retrieve the parameters annotations of `method` using the Jandex API. If you know how to do that from a `MethodInfo` object, I'm all ears!